src/sagemaker/jumpstart/curated_hub/accessors/filegenerator.py

```diff
@@ -0,0 +1,112 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""This module contains important utilities related to HubContent data files."""
+from __future__ import absolute_import
+from functools import singledispatchmethod
+from typing import Any, Dict, List, Optional
+
+from botocore.client import BaseClient
+
+from sagemaker.jumpstart.curated_hub.accessors.fileinfo import FileInfo, HubContentDependencyType
+from sagemaker.jumpstart.curated_hub.accessors.objectlocation import S3ObjectLocation
+from sagemaker.jumpstart.curated_hub.accessors.public_model_data import PublicModelDataAccessor
+from sagemaker.jumpstart.types import JumpStartModelSpecs
+
+
+class FileGenerator:
+    """Utility class to help format HubContent data files."""
+
+    def __init__(
+        self, region: str, s3_client: BaseClient, studio_specs: Optional[Dict[str, Any]] = None
+    ):
+        self.region = region
+        self.s3_client = s3_client
+        self.studio_specs = studio_specs
+
+    @singledispatchmethod
```

Comment:
I haven't seen this annotation before, I assume it's lifted from `aws s3 sync`. Can we use a simpler implementation without these fancy annotations? Unless they're absolutely necessary, I feel like they make maintainability more difficult.

Reply:
Thanks for the input I'll take a look at this. I wanted to take in a single input field that could be one of two types. Realistically, I can have two optional input fields and assert that at least one must be defined. Then I can branch my logic